src/persistent_state.py

```diff
@@ -13,13 +13,17 @@ def load_state_from_disk() -> None:
     """Read the bot state from disk and store it in memory."""
     global _bot_state
 
-    with open(bot_state_file) as f:
-        bot_state_str = f.read()
-
-        if bot_state_str:
-            # Load the JSON representation of the bot's state and convert it to a Python dict so
-            # it can be easily manipulated.
-            _bot_state = json.loads(bot_state_str)
+    bot_state_str = None
+    try:
+        with open(bot_state_file) as f:
+            bot_state_str = f.read()
+    except (FileNotFoundError, IOError):
+        print(f"Could not read state from {bot_state_file}. Continuing...")
```

Comment:
I think we shouldn't print a user-visible error in the case of the `bot_state.json` file not existing, as that will be the state that everyone experiences on first run (or upon setting a custom location).

I think we shouldn't continue on if we fail to read the file as well, as the bot will pretty quickly overwrite the contents of the file with mostly-empty data. I think in that case we should fail loudly and print the exception.

```suggestion
    except FileNotFoundError:
        # Expected on first run or after setting a new custom bot state filepath.
        pass
    except IOError:
        print(f"Could not read state from {bot_state_file}")
        raise
```

Reply:
I was thinking of it as more of a log message than an error, but yeah it's probably not information which is terribly important